### PR TITLE
feat: add xc-tutor.c

### PR DESCRIPTION
**Pull request by lucia-lau**
test

### CODE REVIEW

**Comment by liwenqiu** [comment]
leave a comment

**Comment by liwenqiu** [comment]
leave a comment

**Comment by lucia-lau** [comment]
pls review this pr

File: xc-tutor.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+int *text,            // text segment
+    *old_text,        // for dump text segment
+    *stack;           // stack
+char *data;           // data segment
+int *pc, *bp, *sp, ax, cycle; // virtual machine registers
+
+// instructions
+enum { LEA ,IMM ,JMP ,CALL,JZ  ,JNZ ,ENT ,ADJ ,LEV ,LI  ,LC  ,SI  ,SC  ,PUSH,
+       OR  ,XOR ,AND ,EQ  ,NE  ,LT  ,GT  ,LE  ,GE  ,SHL ,SHR ,ADD ,SUB ,MUL ,DIV ,MOD ,
+       OPEN,READ,CLOS,PRTF,MALC,MSET,MCMP,EXIT };
```
Comment (liwenqiu) [inline xc-tutor.c:23]: first comment

File: xc-tutor.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        return -1;
+    }
+    if (!(stack = malloc(poolsize))) {
+        printf("could not malloc(%d) for stack area\n", poolsize);
+        return -1;
+    }
+
+    memset(text, 0, poolsize);
+    memset(data, 0, poolsize);
+    memset(stack, 0, poolsize);
```
Comment (liwenqiu) [inline xc-tutor.c:146]: second comment

File: xc-tutor.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    text[i++] = IMM;
+    text[i++] = 10;
+    //text[i++] = PUSH;
+    //text[i++] = IMM;
+    //text[i++] = 20;
+    //text[i++] = ADD;
+    //text[i++] = PUSH;
+    text[i++] = EXIT;
+
+    pc = text;
```
Comment (liwenqiu) [inline xc-tutor.c:161]: lgtm

**Review by liwenqiu — CHANGES_REQUESTED** [review]

File: xc-tutor.c (hunk 1 above)
Comment (lucia-lau) [reply xc-tutor.c:23]: reply 1

File: xc-tutor.c (hunk 2 above)
Comment (lucia-lau) [reply xc-tutor.c:146]: reply 2

File: xc-tutor.c (hunk 3 above)
Comment (lucia-lau) [reply xc-tutor.c:161]: reply 3

**Review by liwenqiu — APPROVED** [review]

**Pull request merged**